### PR TITLE
feat: add release files

### PR DESCRIPTION
**Pull request by andrewrynhard**
This adds the hack files for generating a release.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

File: hack/git-chglog/CHANGELOG.tpl.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# THIS FILE WAS AUTOMATICALLY GENERATED, PLEASE DO NOT EDIT.
```
Comment (andrewrynhard) [inline hack/git-chglog/CHANGELOG.tpl.md:1]: I need to add a way to comment for markdown `<!-- -->`

File: internal/output/preamble.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ var PreambleTimestamp time.Time
 var PreambleCreator string
 
 // Preamble returns file auto-generated preamble with specified comment character.
-func Preamble(commentPrefix string) string {
+func Preamble(commentPrefix string, commentPostFixes ...string) string {
```
Comment (andrewrynhard) [inline internal/output/preamble.go:22]: Made this variadic to make the postfix optional. Don't know of a cleaner way other than introducing functional opts. Happy to make another PR for that.

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # THIS FILE WAS AUTOMATICALLY GENERATED, PLEASE DO NOT EDIT.
 #
-# Generated on 2020-08-12T20:46:00Z by kres 4718524-dirty.
+# Generated on 2020-08-18T04:07:32Z by kres 66eb135-dirty.
```
Comment (andrewrynhard) [inline Makefile:3]: I don't understand why this change happens, but not for other files like `Dockerfile` and `.gitignore`.

File: hack/release.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# THIS FILE WAS AUTOMATICALLY GENERATED, PLEASE DO NOT EDIT.
```
Comment (andrewrynhard) [inline hack/release.sh:1]: I could use some ideas on how we should mark this script as executable.

File: Makefile (hunk 3 above)
Comment (smira) [reply Makefile:3]: I guess at some point file got changed and probably change got reverted back, but the preamble got updated.

File: internal/output/release/release.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+		f = o.release
+	case config:
+		f = o.config
+	case template:
+		f = o.template
+	default:
+		panic("unexpected filename: " + filename)
+	}
+
+	if err := os.MkdirAll(filepath.Dir(filename), 0o755); err != nil {
```
Comment (smira) [inline internal/output/release/release.go:177]: I guess this could be moved to the `FileAdapter` (creating required directories), it shouldn't hurt in general

File: internal/output/release/release.go [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+		return err
+	}
+
+	if err := f(w); err != nil {
+		return err
+	}
+
+	if filename == release {
+		return os.Chmod(filename, 0o744)
```
Comment (smira) [inline internal/output/release/release.go:186]: I'm not sure what's the best way in general here, but at the moment `GenerateFile` is called file might not exist yet (it's buffered for content comparison).

Probably same way, provide some interface to return default file mode on creation (optional interface?)

File: internal/output/release/release.go (hunk 5 above)
Comment (andrewrynhard) [reply internal/output/release/release.go:177]: :+1: agree

File: internal/output/release/release.go (hunk 6 above)
Comment (andrewrynhard) [reply internal/output/release/release.go:186]: Yeah, I wasn't sure what the best way was. I will dig into it some more.

Comment (andrewrynhard) [reply internal/output/release/release.go]: Done.

**Review by smira — APPROVED** [review]
